### PR TITLE
Fix NXBT-1292 upgrade list

### PR DESCRIPTION
**Pull request by yjulienne-nuxeo**

### CODE REVIEW

File: nuxeo-connect-client/src/main/java/org/nuxeo/connect/packages/PackageManagerImpl.java [hunk 1]
```diff
@@ -427,6 +426,30 @@ public void registerSource(PackageSource source, boolean local) {
...
+        return upgradeNames;
+    }
+
+    @Override
+    public List<String> listHotfixesNames(String targetPlatform, boolean allowSNAPSHOT) {
+        List<DownloadablePackage> hotFixes = listPackages(PackageType.HOT_FIX, targetPlatform);
+        List<String> hotFixNames = new ArrayList<>();
+        hotFixes.forEach(pkg -> {
+            if(!hotFixNames.contains(pkg.getName()) && (!pkg.getVersion().isSnapshot() || allowSNAPSHOT)){
+                hotFixNames.add(pkg.getName());
```
Comment (kevinleturc) [inline nuxeo-connect-client/src/main/java/org/nuxeo/connect/packages/PackageManagerImpl.java]: Here, it's better to use stream().filter(pkg -> ...).map(...).collect(Collectors.toList()); as it there's no possible side effect as you construct the new list and not modify an external one. (http://stackoverflow.com/questions/28319064/java-8-best-way-to-transform-a-list-map-or-foreach)

File: nuxeo-connect-client/src/main/java/org/nuxeo/connect/packages/PackageManagerImpl.java [hunk 2]
```diff
@@ -427,6 +426,30 @@ public void registerSource(PackageSource source, boolean local) {
     }
 
     @Override
+    public List<String> listInstalledPackagesNames(PackageType pkgType) {
+        List<DownloadablePackage> upgrades = listInstalledPackages();
+        List<String> upgradeNames = new ArrayList<>();
+        for (DownloadablePackage upgrade : upgrades) {
```
Comment (kevinleturc) [inline nuxeo-connect-client/src/main/java/org/nuxeo/connect/packages/PackageManagerImpl.java]: Here you can also use stream as in listHotfixesNames

File: nuxeo-connect-client/src/main/java/org/nuxeo/connect/packages/PackageManagerImpl.java [hunk 3]
```diff
@@ -480,58 +503,19 @@ public void registerSource(PackageSource source, boolean local) {
...
-                    }
-                    if (!alreadyInLocal) {
-                        toUpdate.add(0, pkg);
-                    }
-                }
-            }
-        }
+        List<String> toUpdateIds = resolution.getOrderedPackageIdsToInstall();
+        List<DownloadablePackage> toUpdate = new ArrayList<>();
+        toUpdateIds.forEach(pkgId -> toUpdate.add(getPackage(pkgId)));
```
Comment (kevinleturc) [inline nuxeo-connect-client/src/main/java/org/nuxeo/connect/packages/PackageManagerImpl.java]: Same thing here about stream() use.

File: nuxeo-connect-client/src/main/java/org/nuxeo/connect/packages/PackageManagerImpl.java [hunk 4]
```diff
@@ -427,6 +426,30 @@ public void registerSource(PackageSource source, boolean local) {
     }
 
     @Override
+    public List<String> listInstalledPackagesNames(PackageType pkgType) {
+        List<DownloadablePackage> upgrades = listInstalledPackages();
```
Comment (jcarsique) [inline nuxeo-connect-client/src/main/java/org/nuxeo/connect/packages/PackageManagerImpl.java]: Just about variable naming: `upgrades` is not consistent with the method name and its behavior.

File: nuxeo-connect-client/src/main/java/org/nuxeo/connect/packages/PackageManagerImpl.java [hunk 5]
```diff
@@ -427,6 +426,30 @@ public void registerSource(PackageSource source, boolean local) {
...
+        }
+        return upgradeNames;
+    }
+
+    @Override
+    public List<String> listHotfixesNames(String targetPlatform, boolean allowSNAPSHOT) {
+        List<DownloadablePackage> hotFixes = listPackages(PackageType.HOT_FIX, targetPlatform);
+        List<String> hotFixNames = new ArrayList<>();
+        hotFixes.forEach(pkg -> {
+            if(!hotFixNames.contains(pkg.getName()) && (!pkg.getVersion().isSnapshot() || allowSNAPSHOT)){
```
Comment (jcarsique) [inline nuxeo-connect-client/src/main/java/org/nuxeo/connect/packages/PackageManagerImpl.java]: check `allowSNAPSHOT` first
`if(!hotFixNames.contains(pkg.getName()) && (allowSNAPSHOT || !pkg.getVersion().isSnapshot()))`

File: nuxeo-connect-client/src/main/java/org/nuxeo/connect/packages/PackageManagerImpl.java (hunk 3 above)
Comment (jcarsique) [reply nuxeo-connect-client/src/main/java/org/nuxeo/connect/packages/PackageManagerImpl.java]: @kevinleturc worth to add a section into https://doc.nuxeo.com/display/CORG/Coding+and+Design+Guidelines

**Review by jcarsique — APPROVED** [review]
:+1:

**Pull request merged**